signature/jws/jwt.go

```diff
@@ -0,0 +1,151 @@
+package jws
+
+import (
+	"crypto"
+	"crypto/x509"
+	"encoding/base64"
+	"fmt"
+
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/notaryproject/notation-core-go/signature"
+)
+
+// signingMethod is the interface for jwt.SigingMethod with additional method to
+// access certificate chain after calling Sign().
+type signingMethod interface {
+	jwt.SigningMethod
+
+	// CertificateChain returns the certificate chain.
+	//
+	// It should be called after calling Sign().
+	CertificateChain() ([]*x509.Certificate, error)
+
+	// PrivateKey returns the private key.
+	PrivateKey() crypto.PrivateKey
+}
+
+// remoteSigningMethod wraps the remote signer to be a SigningMethod
+type remoteSigningMethod struct {
+	signer    signature.Signer
+	certs     []*x509.Certificate
+	algorithm string
+}
+
+func newRemoteSigningMethod(signer signature.Signer) (signingMethod, error) {
+	algorithm, err := extractJwtAlgorithm(signer)
+	if err != nil {
+		return nil, err
+	}
+	return &remoteSigningMethod{
+		signer:    signer,
+		algorithm: algorithm,
+	}, nil
+}
+
+// Verify doesn't need to be implemented.
+func (s *remoteSigningMethod) Verify(signingString, signature string, key interface{}) error {
+	panic("not implemented")
+}
+
+// Sign hashes the signingString and call the remote signer to sign the digest.
+func (s *remoteSigningMethod) Sign(signingString string, key interface{}) (string, error) {
+	// sign by external signer
+	sig, certs, err := s.signer.Sign([]byte(signingString))
+	if err != nil {
+		return "", err
+	}
+	s.certs = certs
+	return base64.RawURLEncoding.EncodeToString(sig), nil
+}
+
+// Alg return the signing algorithm.
+func (s *remoteSigningMethod) Alg() string {
```

Comment:
Let's use the full name of Algorithm

Reply:
Added a coments to mention it implements jwt.SigningMethod interface
'''
type SigningMethod interface {
	Verify(signingString, signature string, key interface{}) error // Returns nil if signature is valid
	Sign(signingString string, key interface{}) (string, error)    // Returns encoded signature or error
	Alg() string                                                   // returns the alg identifier for this method (example: 'HS256')
}
'''